edx_analytics_transformers/routers/requests_router.py

```diff
@@ -0,0 +1,165 @@
+"""
+Generic router to send events to hosts.
+"""
+import logging
+
+from eventtracking.processors.exceptions import EventEmissionExit
+
+from edx_analytics_transformers.utils.http_client import HttpClient
+from edx_analytics_transformers.django.models import RouterConfiguration
+
+
+logger = logging.getLogger(__name__)
+
+
+ROUTER_STRATEGY_MAPPING = {
+    'AUTH_HEADERS': HttpClient,
+}
+
+
+class RequestsRouter:
+    """
+    Router to send events to hosts using requests library.
+    """
+
+    def __init__(self, processors=None, backend_name=None):
+        """
+        Initialize the router.
+
+        Arguments:
+            processors (list):   list of processor instances
+            backend_name (str):  name of the router backend
+        """
+        self.processors = processors if processors else []
+        self.backend_name = backend_name
+
+    def send(self, original_event, transformed_event):
+        """
+        Send the event to configured routers after processing it.
+
+        Event is processed through the configured processors. A router config
+        object matching the backend_name (and that was modified last) is used
+        to get the list of hosts to which the event is required to be delivered to.
+        `requests` module is then used to send the event to such hosts.
+
+        Arguments:
+            original_event (dict):      original event dictionary
+            transformed_event (dict):   transformed event dictionary
+        """
+        try:
+            logger.info(
+                'Processing event %s for router with backend %s',
+                original_event['name'],
+                self.backend_name
+            )
+
+            processed_event = self.process_event(transformed_event)
+        except EventEmissionExit:
+            logger.info(
+                'Could not process event %s for backend %s\'s router',
+                original_event['name'],
+                self.backend_name,
+                exc_info=True
+            )
+            return
+
+        logger.info('Successfully processed event %s for router with backend %s',
```

Comment:
How noisy is this going to be? Should this be a `debug` message instead of an `info` message?